tests/sentry/search/events/builder/test_metrics.py

```diff
@@ -2055,54 +2094,89 @@ def test_query_normal_distribution(self):
 
 
 class AlertMetricsQueryBuilderTest(MetricBuilderBaseTest):
+    def test_standard_metrics(self):
+        query = AlertMetricsQueryBuilder(
+            self.params,
+            use_metrics_layer=False,
+            granularity=3600,
+            query="",
+            dataset=Dataset.PerformanceMetrics,
+            selected_columns=["p75(measurements.fp)"],
+            on_demand_metrics_enabled=True,
+        )
+
+        assert query.on_demand_metrics_enabled
+        assert not query._on_demand_spec
+
+    def test_run_on_demand_query(self):
+        query = AlertMetricsQueryBuilder(
+            self.params,
+            use_metrics_layer=False,
+            granularity=3600,
+            query="transaction.duration:>=100",
+            dataset=Dataset.PerformanceMetrics,
+            selected_columns=["p75(measurements.fp)"],
+            on_demand_metrics_enabled=True,
+        )
+
+        result = query.run_query("test_query")
+
+        assert len(result["data"]) == 1
+
+        meta = result["meta"]
+
+        assert len(meta) == 2
```

Comment:
There should be an `is_metrics_data` flag in there, can you assert this and check if the current way it's set makes sense?

Reply:
Unfortunately, that flag is always set by the callers of query builders, after they receive results such as in https://github.com/getsentry/sentry/blob/096b21418cd3e6ebc778da255ca0c66679c04672/src/sentry/snuba/metrics_performance.py#L67 
The result itself here contains only "meta" and "data"